app/helpers/publishers_helper.rb

```diff
@@ -19,11 +19,19 @@ def publisher_can_receive_funds?(publisher)
     publisher.uphold_status == :verified
   end
 
-  def publisher_humanize_balance(publisher, currency)
-    if balance = publisher.wallet &&
-        publisher.wallet.contribution_balance.is_a?(Eyeshade::Balance) &&
-        publisher.wallet.contribution_balance
-      '%.2f' % balance.convert_to(currency)
+
+  def next_deposit_date(today = DateTime.now)
+    if today.day > 8
+      today = today + 1.month
+    end
+    today.strftime("%B 8th")
+  end
+
+  def publisher_overall_bat_balance(publisher)
```

Comment:
Since we're refactoring this I think that it would be cleaner create an ActiveModel class for wallet. That would remove the need for having all these helpers which are scoped to `publisher.wallet`. 

I was thinking something like this. 

https://gist.github.com/corymcdonald/97914936a2f8b70e997e6a49a0f76e1a

Then in the `model/publisher.rb` file we could have something like this for the wallet
```ruby
  # API call to eyeshade
  def wallet
    @wallet ||= Wallet.new(self)
    @wallet = nil if @wallet.eyeshade_wallet.nil?
    @wallet
  end
```

However, it might not be worth creating an extra class just for the sake of isolation. 🤔